server/fleet/app.go

```diff
@@ -420,10 +420,19 @@ func (s *MacOSSettings) FromMap(m map[string]interface{}) (map[string]bool, erro
 
 // MacOSSetup contains settings related to the setup of DEP enrolled devices.
 type MacOSSetup struct {
-	BootstrapPackage            optjson.String `json:"bootstrap_package"`
-	EnableEndUserAuthentication bool           `json:"enable_end_user_authentication"`
-	MacOSSetupAssistant         optjson.String `json:"macos_setup_assistant"`
-	EnableReleaseDeviceManually optjson.Bool   `json:"enable_release_device_manually"`
+	BootstrapPackage            optjson.String                     `json:"bootstrap_package"`
+	EnableEndUserAuthentication bool                               `json:"enable_end_user_authentication"`
+	MacOSSetupAssistant         optjson.String                     `json:"macos_setup_assistant"`
+	EnableReleaseDeviceManually optjson.Bool                       `json:"enable_release_device_manually"`
+	Script                      optjson.String                     `json:"script"`
+	Software                    optjson.Slice[*MacOSSetupSoftware] `json:"software"`
```

Comment:
Just curious, why does this need to be a pointer: `*MacOSSetupSoftware`?

Reply:
Not strictly needed, but makes it easier to avoid issues as we need to change its field value here: https://github.com/fleetdm/fleet/blob/c11420b12df3ecd8bdb732736b138ae4b1aa38df/server/service/client.go#L858

A non-pointer would only change the local copy and would turn this into a potentially tricky bug to find.